src/main/resources/common-services/CDAP/5.0.0/package/scripts/master.py

```diff
@@ -66,56 +66,63 @@ def killOlderProcesses(self):
-                                                     status, output = commands.getstatusoutput("tail -n1 /tmp/twill_" + service + "_instances_output")
-                                                     if status == 0:
-                                                         print(output)
-                                                         data = json.loads(output)
-                                                         container_id = data['data']['containerId']
-                                                         container_values = container_id.split('_')
-                                                         applicationId = "application_" + container_values[2] + "_" + container_values[3] 
-                                                         kinit_cmd = params.kinit_cmd_master
-                                                         kill_application_cmd = format("{kinit_cmd} yarn application -kill " + applicationId)
-                                                         print(kill_application_cmd)
-                                                         Execute(kill_application_cmd, user='cdap')
-                                                         print("applicationId : " + applicationId)
-                                                     Execute("rm -rf /tmp/twill_" + service + "_instances_output", user='zookeeper') 
-                                         Execute("rm -rf /tmp/twill_" + service + "_instances", user='zookeeper') 
-                    Execute("rm -rf /tmp/twill_master_services", user='zookeeper')                  
+        if status != 0:
+            print('fail in executing the command ls /election/master.services on zookeeper')
+            return
+        if (!(output.startswith('[') and output.endswith(']') and output == '[]')):
+            print('outt=put value is not equal to []')
+            return
+        # exectute ls /cdap/twill and save its output
+        Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " ls /twill > /tmp/twill_master_services", user='zookeeper')
+        status, output = commands.getstatusoutput("tail -n1 /tmp/twill_master_services")
+        if status != 0:
+            print('fail in executing the command ls /twill on zookeeper')
+            return
+        if (!(output.startswith('[') and output.endswith(']') and output != '[]')):
+            print('output value is equal to []')
+            return
+        print(output)
+        result = output[1:len(output)-1]
+        services = result.split(',')
+        # iterate over last output and execute ls /cdap/twill/<name>/instances and save its output
+        for service in services:
+            service = service.strip()
+            if (service != 'master.services'):
```

Comment:
This 'if' block can be removed by doing something like:
if service == 'master.services':
    continue

Reply:
added continue